packages/loader/container-loader/src/loader.ts

```diff
@@ -392,6 +399,11 @@ export class Loader extends EventEmitter implements ILoader {
                     resolvedAsFluid);
         }
 
+        if (newlyLoaded && this.services.options.readonly === true) {
+            // Don't re-force readonly when returning an existing container
```

Comment:
It should be no-op if container is already read-only.
Note that if someone is manipulating this container and reset its read-only state, and now someone else is asking for container and expects it to be read-only, then we are breaking that promise.

I think the only way to make it work correctly is by
- providing a logic (interface/callback/...) to loader instance at loader creation
- This logic control all containers in the loader in terms of things like read-only, potentially pausing, etc.
In other words there is agreed upfront policy that applies to all containers in loader.

This works well with read-only, and clients use it to disallow user from making changes on loss of connection. So that logic is host policy for all containers.

Note that "all containers' here is very important. If we have guest components in container (that is component that is coming from another container), then first container will load second container. Host is not aware about it, and first container is not aware of host policies. When host says - container should be read-only, it really means - this container and all nested containers as well. Today there is no way to represent that logic, thus my point about it being a policy (expressed in code provided to loader) and applicable to all containers in loader.
If someone wants to escape that policy, it creates new loader with new policy (might be applicable for summarizer, for example).

This can't be solved with options.
I think (though I have not had enough time to think deeply about it) that pausing container is in the same bucket - it's a policy on how we load stuff. It likely as important for nested containers as well.

Reply:
To clarify, I believe that
1. policy should be applicable to all containers in loader, with no way to override it per container request.
2. Some policies can be expressed as boolean, but most can't. forceReadonly and paused are two examples - their initial state is not interesting, what's interesting is that policy is applied uniformly through lifetime of containers, and requesting an instance of container from loader at any moment gives expected result (i.e. using Bohemia policy - if container still did not load fully - it is paused, if it already loaded and rendered, it's no longer paused).

Reply:
I opened a new issue #4880 to track migrating the non-cache policy into the loader and remove it from the request and captured your thoughts there.  That work involves some changes with how we handle creating the scheduler and summarizer and potentially has implications for decoupling Loader from container, so I think it probably reaches too far beyond the caching issue